### ISSUE TITLE
Namespaces dependent on solution folder name?

### ISSUE THREAD

**Issue by benhysell** [aspnet/Blazor#854]
- Blazor 0.3.0
- VS 2017 15.7.1

Created a new solution via PowerShell:
```c:\projects\NewProject\ dotnet new blazor```

Created a new project in c:\projects\CoolNewProject called CoolNewProject.

Normally I like to have my .sln file a directory above my projects, i.e.
```c:\projects\CoolNewProject\CoolNewProject.sln```

Then throw all of my src in 'src'.
```c:\projects\CoolNewProject\src\CoolNewProject.csproj``` and all of my other project files.

The second I move everything into the 'src' folder from my newly created Blazor project I can no longer compile:
```
Error | CS0246 | The type or namespace name 'App' could not be found (are you missing a using directive or an assembly reference?) | CoolNewProject | C:\projects\CoolNewProject\src\Program.cs | 17 | Active
```

Additionally, none of the Pages can find their namespaces either, i.e.
```
Error | CS0234 | The type or namespace name 'Shared' does not exist in the namespace 'CoolNewProject' (are you missing an assembly reference?) | CoolNewProject | C:\Projects\CoolNewProject\src\obj\Debug\netstandard2.0\Pages\Counter.g.i.cs | 21 | Active
```

**Comment by Andrzej-W** [comment]
I created a new CoolNewProject (Blazor hosted) in Visual Studio and closed VS. I moved all three projects into src subfolder and manually changed projects' paths in CoolNewProject.sln. Then I loaded solution in VS, compiled and everything works perfectly.

I repeated the above steps in console window and everything works as expected:

- dotnet new blazorhosted
- create src subfolder and move all three projects there
- manually **edit paths in solution file**
- dotnet build (in main CoolNewProject folder)
- change directory to src\CoolNewProject.Server
- dotnet run
- start browser and connect to localhost:5000

**Comment by benhysell** [comment]
I agree ```dotnew new blazorhosted``` works as expected

```botnet new blazer``` does not

**Comment by Andrzej-W** [comment]
I can confirm your observation - it is impossible to compile project created with `dotnet new blazor` after moving to `src` subfolder.

Originally I tested `blazorhosted` because you mentioned about sln file and `blazor` project is created without solution file. I don't know why but is probably somehow related to the fact that we have or we don't have a solution file.

**Comment by benhysell** [comment]
Update, just tried this in 0.4.0 release, issue remains.

**Comment by halkar** [comment]
I can reproduce this issue as well.

**Comment by DNF-SaS** [comment]
Reproduction-Steps:

1) Create new Blazor project "BlazorNamespaceTest" from VS-Template (using standalone "Blazor"-Template)
2) Edit the corresponding BlazorNamespaceTest.sln-File - change the Project-Reference (e.g. incorrect casing)
  `Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "BlazorNamespaceTest", "BlazorNameSPACETest\BlazorNamespaceTest.csproj", "{B6D9B760-70C0-4361-B78A-4C60CB1332FD}"`
3) Building the project will fail
4) The generated app.g.i.cs contains the wrong namespace "BlazorNameSPACETest"

**Comment by mokeyish** [comment]
Generated  namespace should depend on `*.csproj` filename or  `RootNamespace` property defined in *.csproj.

**Comment by rynowak** [comment]
Yeah this a limitation we have right now unfortunately. I think we all agree that using `RootNamespace` would be the correct thing to do. This will require changes to the Razor SDK and VS to support, so its a few months away.
